CONTRIBUTING.md

```diff
@@ -0,0 +1,100 @@
+5. Implement your changes along with relevant tests for the issue. Please make sure you are covering unit-tests where required.
+6. Create a pull request against **master** branch.
+
+
+
+## Code Style
+
+We follow the **reStructuredText** docstring format (default of PyCharm), along with typing.
+```python
+def python_function(first_argument: int, second_argument: int) -> str:
+    """Do something with the two arguments.
+
+    :param first_argument: First argument to the function
+    :param second_argument: Second argument to the function
+    :return: Description of the output
+    """
+```
+
+
+
+
+## Code Formatting
+
+We enforce [black](https://github.com/psf/black) code formatting in addition to existing [flake8](https://flake8.pycqa.org/en/latest/user/index.html) checks. 
+
+To ensure everyone uses same code style, a project-wise [configuration file](https://github.com/Deci-AI/data-gradients/blob/master/pyproject.toml) has been added to DG repo. It ensures all formatting will be exactly the same regardless of OS, python version or the place where code formatting check is happening. 
+
+### Installation
+
+To start, one need to install required development dependencies (actual versions of black, flake8 and git commit hooks):
+
+`$ pip install -r requirements.dev.txt`
+
+### Pre-Commit Hooks
+
+A pre-commit hook as an easy way to ensure all files in the commit are already formatted accordingly and pass linter checks. If they are not, the git will prevent commit of problematic files unless errors are fixed.
+
+To start, run the following command from DG repo root:
+```bash
+$ pip install pre-commit
+$ pre-commit install
+```
+
+The command should complete without errors. Once done, all your upcoming commits will be checked via black & flake8. 
+
+### Usage
+
+Just run ```$ black .``` from the DG root. It will reformat the whole repo.
+For flake8: ```$ flake8 --statistics --config scripts/flake8-config setup.py .``` 
+
+
+## Signed Commits
+
+### Background
+
+Signed commits provide a way to verify the authenticity and integrity of the code changes made by a particular developer, as the commit is cryptographically signed using their private GPG key. This helps ensure that the code changes were made by the intended person and have not been tampered with during transit.
+
+You can find more information [here](https://withblue.ink/2020/05/17/how-and-why-to-sign-git-commits.html).
+
+### Add GPG key to GitHub
+
+1. [Generate a new GPG key](https://docs.github.com/en/authentication/managing-commit-signature-verification/generating-a-new-gpg-key) 
+2. Copy the GPG key by running the command on step 12 from the link above
+    
+```bash
+$ gpg --armor --export 3AA5C34371567BD2
+```
+    
+3. [Add the new GPG key to your GitHub account](https://docs.github.com/en/authentication/managing-commit-signature-verification/adding-a-new-gpg-key-to-your-github-account)
```

Comment:
Do we also want to introduce signed commits here ?

Reply:
yes. why not...

Reply:
It's extra work for external contributors, and it can be intimidating for those who have a simple change they want to contribute.
I'm not fundamentally against but I'm wondering if eventually it is really useful, many big open-source projects are not using this. But maybe they are wrong and we should still keep it, I am just wondering

Reply:
ok. for now, let's not force it in the setting, but keep it here in the guidelines as "best practice"

Reply:
Ok